### PR TITLE
Pehapkari/gulp docker

### PR DESCRIPTION
**Pull request by newPOPE**
Hi @TomasVotruba, 

here is PR for docker and gulp. 

- rewritten Dockerfile (optimized)
- add nodejs and deps
- bind php dev server to 0.0.0.0

but there is an issue with gulp watch, it seems that watching not work in docker. I will try it later again.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ gulp.task('default', function () {
     });
 
     // Run local server, open localhost:8000 in your browser
-    exec('php -S localhost:8000 -t output');
+    exec('php -S 0.0.0.0:8000 -t output');
```
Comment (TomasVotruba) [inline gulpfile.js]: Why this change? It's much less readable for me. 
Is it broken without it? If so: `127.0.0.1` might be better.

**Comment by TomasVotruba** [comment]
Hey @newPOPE , thank you a lot for this.

File: gulpfile.js (hunk 1 above)
Comment (newPOPE) [inline gulpfile.js]: I don't think so. `127.0.0.1` is the same as localhost in `/etc/hosts` in your host machine. It (maybe) can work on Linux systems but I'am using OSX and docker via VM and binding must be like I wrote.

Comment (TomasVotruba) [inline gulpfile.js]: Yea, that is localhost. That's the goal. Could you describe the issue? 
Could it be solved in dockerfile itself?

It would be better, because this would influence many posts and gulp files over other websites.

Comment (newPOPE) [inline gulpfile.js]: It is like bind IP for mysql. When you setup mysql for bind 127.0.0.1 or localhost you can connect only from the same machine to mysql not from other. That's a reason why `0.0....` must be there.

What you mean posts and websites?

You can ping me on slack as `new_pope`

Comment (TomasVotruba) [inline gulpfile.js]: > What you mean posts and websites?

It would be inconsistent, ergo WTF the same way is for me now, with other websites with Statie and `gulpfile.js`.

By post I Mean: https://www.tomasvotruba.cz/blog/2017/02/20/statie-how-to-run-it-locally/#minitip-use-gulp-work-for-you

Slack it is...

Comment (TomasVotruba) [inline gulpfile.js]: @Majkl578 I'm not sure, but do you have OSX? If so, did you came across the same issue?

Comment (TomasVotruba) [inline gulpfile.js]: We resolved it with @newPOPE on Slack and it's ok this way.

The url `localhsts:8000` remains the same, so it's still user friendly as I wanted.

Free free to merge @newPOPE , after you resolve the watch issue you metioned.

**Comment by newPOPE** [comment]
@TomasVotruba I merged this to this repo&branch. I rewrite `gulpfile` a little for watch files. Can you check that on your machine (without docker) https://github.com/pehapkari/pehapkari.cz/tree/gulp-docker. 

Thanks

File: gulpfile.js (hunk 1 above)
Comment (Majkl578) [inline gulpfile.js]: I have Linux, but I am 👎 on changing this to 0.0.0.0, since it exposes the instance globally to the world when not run under docker... I understand it doesn't work under docker, but it should be solved differently.

Comment (newPOPE) [inline gulpfile.js]: @Majkl578 we can discuss here #236. We talk with @TomasVotruba about that change. And finally he found https://github.com/Symplify/Statie/blob/v1.3.0/src/HttpServer/HttpServer.php#L83 :)

**Comment by TomasVotruba** [comment]
@newPOPE Looks ok to me. Good idea do add `packages.json`.

Could you also update the README please? https://github.com/pehapkari/pehapkari.cz/blob/master/README.md#install
